[Bug] Show static board results

theBeginner86 opened the pull request:
**Description**

This PR adds the static board result on performance graph charts.

It was removed because:
-  this [function](https://github.com/meshery/meshery/blob/master/ui/components/MesheryChart.js#L32) wasn't destructing the `Kubernetes` key having the data for `nodes` and `Kubernetes server version` in the form of nested structure.
- [Here](https://github.com/meshery/meshery/blob/master/ui/lib/chartjs-formatter.js#L147) we were explicitly hiding the value

Now I'm passing the each `node's data` and `k8s server version` after destructuring so that it can be shown in the desired way

This PR fixes #

**Notes for Reviewers**

Demo:

https://user-images.githubusercontent.com/73700530/165699801-6200384d-f5bd-4737-b5d5-98a086575323.mp4




**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

warunicorn19 commented:
I think we need to find a better way to represent this data. When there's just 1 node it still looks alright but with multiple nodes, it doesn't look great. And is the value of the container runtime correct?

theBeginner86 commented:
@warunicorn19 
1. Ok. I too thought that but to align it with what it used to be I made it look same.
Though, I can make it look like this as well:

<img width="1289" alt="Screenshot 2022-04-28 at 2 35 35 PM" src="https://user-images.githubusercontent.com/73700530/165718321-6dc25fdc-d947-4739-a825-abad40b82aa1.png">
<img width="1440" alt="Screenshot 2022-04-28 at 2 35 06 PM" src="https://user-images.githubusercontent.com/73700530/165718330-54cf7c67-c689-404a-8bc0-9a4aed96f895.png">

But I think this way, it only looks good for multiple nodes only. 
Though please let me know if there could be some other better way to show this info 😅


2. I think that container runtime provides the version for docker-engine on the host machine. But I'm not sure.
<img width="1289" alt="Screenshot 2022-04-28 at 2 38 12 PM" src="https://user-images.githubusercontent.com/73700530/165718757-a159f8dd-9426-4bfd-8a5a-3e7f762b79ac.png">

leecalcote commented:
@gyohuangxin

leecalcote commented:
Good work in brining this back, @theBeginner86. In either case of a single or multiple nodes, the prevention of cluster information needs to be done differently.  

@mchurichi @jtaleric @sunkuranganath your perspectives here would be great to have.

gyohuangxin commented:
@theBeginner86 Great job! Can we present node's information as a table like this:
![image](https://user-images.githubusercontent.com/42127654/165878854-1f506887-28e9-4757-b41a-7d3ed1b20c2f.png)
I think it's more intuitive  whether there is one or multiple nodes.

Chadha93 commented:
@theBeginner86 @leecalcote @gyohuangxin A very quick design that could work, keeping the graphical representation aspect ratio as highlight with the nodes table. 
![static board performance](https://user-images.githubusercontent.com/65453826/165939826-09054c68-9e4d-4395-887d-9555507fcb4d.png)

theBeginner86 commented:
Thanks, @gyohuangxin, and @Chadha93. These are great suggestions.

I'll work on these suggested changes 👍

mchurichi commented:
Hey @theBeginner86, this looks neat! it's definitely a nice addition :)

I just have two questions:

1. I'm not a Meshery expert, but my understanding is that it can be deployed either in-cluster or out of it, so I'm wondering, how is this information relevant for the latter?

2. Why is the OS showing up as `Docker Desktop` here? it seems to be the node image, not the OS itself. I think there's a different node property that shows the OS (`linux`, `darwin`, etc).

leecalcote commented:
Some great feedback here. 👏 

@mchurichi good question regarding #1 above. As it turns out, in either deployment scenario, users interpreting the performance test results are likely to want to review details of the environment in which the performance test was run. Whether Meshery is deployed on it's own docker host or deployed inside the Kubernetes cluster, it's of value to capture the cluster details and have those available for review, so that the performance can be assessed in context of the environment used and the environment's resources. 

At this point, I'm saying things that you already know, but am just taking a moment to spell them out a bit.

theBeginner86 commented:
@mchurichi 
These are great questions!


2. Yeah you are correct 😅. From the backend instead of sending the property `operating_system`, we are currently sending `os_image`. So, we are seeing **Docker Desktop**. After changing that, it would be **linux**.
Thanks! Good eye.

<img width="1440" alt="Screenshot 2022-04-06 at 12 29 14 AM" src="https://user-images.githubusercontent.com/73700530/166264436-b8db485f-043b-42df-83ab-b841e2192830.png">

jtaleric commented:
> @theBeginner86 @leecalcote @gyohuangxin A very quick design that could work, keeping the graphical representation aspect ratio as highlight with the nodes table. ![static board performance](https://user-images.githubusercontent.com/65453826/165939826-09054c68-9e4d-4395-887d-9555507fcb4d.png)

I wonder if we should have a separate page for the SUT details if the SUT is > N nodes?  To help cleanup the interface.

mchurichi commented:
@leecalcote @theBeginner86
Thanks to you both for your replies. As for question # 1 about cluster info, just to be clear, I'm not suggesting any change, just wanted to know if there was something else that someone doing a perf test needs to be aware of when running in those 2 different deployment scenarios.

@leecalcote we talked a few weeks ago about cluster metrics while running performance tests, and this seems to be pretty aligned with that.
Can't wait to see this merged!

leecalcote commented:
@mchurichi, oh, I see. Yes, you're right. Whether a given performance test was run in deployment scenarios in which Meshery's load generator is local to the node or cluster in which the system under test is running or not, is pertinent context for viewers of the results to have. There are a couple of considerations to make here. Perhaps, a simple starting point is recording the platform and location of where Meshery's load generator was running at the time of the test (as opposed to building in logic to ascertain whether that location is cluster or node-local to the SUT).

@jtaleric, what a great suggestion. Yes, pagination of the table of nodes and clusters is necessary as well as having this paginated tabular output displayed on a tab/page of the report makes much sense.

theBeginner86 commented:
**This PR is ready for review and merge**

After new changes:

https://user-images.githubusercontent.com/73700530/167105946-051d3fd3-90df-44e0-a673-5ff351dbbe6d.mov


Updates:
- [x] Created a new component for this node details to handle, manage and maintain it easily.
- [x] Added [MUI datatables](https://github.com/gregnb/mui-datatables) for tabulations.
- [x] Support for pagination.
- [x] Updated scale for the memory metadata. Changed Kibibyte Ki ---> Gigabyte Gi. Conversion rate: **1 Ki =0.000001024 Gi**. (**Please someone once confirm if it's giving the right result**).
- [x] Updated the node property to use `operating_system`.

warunicorn19 commented on ui/components/NodeDetails.js:
```diff
@@ -0,0 +1,206 @@
+import MUIDataTable from "mui-datatables";
+import TableSortLabel from '@material-ui/core/TableSortLabel'
+
+class NodeDetails extends React.Component {
```
Any reason why we're using a class based component here instead of a functional component?

warunicorn19 replied:
If it's not too much of a hassle please let's switch to a functional component😅.

warunicorn19 reviewed (commented):
Changes look pretty neat. Good stuff! @theBeginner86

theBeginner86 commented:
https://user-images.githubusercontent.com/73700530/167273271-49d52a41-8801-4fac-b603-662e1f434641.mov

Updates:
- [x] Changed class component to functional component.
- [x] Created a separate modal for node details

gyohuangxin commented:
@theBeginner86 Excellent job!  A separate modal for node details looks good for me.

warunicorn19 reviewed (approved):
Lgtm!

gyohuangxin commented:
@theBeginner86 I'm interested with the "download" and "print" buttons, what will happen after clicking them? 
![image](https://user-images.githubusercontent.com/42127654/167335634-5a0df8b5-8d31-4ca5-9586-0f37e6ac6c0f.png)

theBeginner86 commented:
> @theBeginner86 I'm interested with the "download" and "print" buttons, what will happen after clicking them?
> ![image](https://user-images.githubusercontent.com/42127654/167335634-5a0df8b5-8d31-4ca5-9586-0f37e6ac6c0f.png)

@gyohuangxin 
Actually, I didn't override the default table settings provided by MUI. So, those buttons come inbuilt with the MUI data-tables, I'm using here.
- Download btn:  downloads the `.csv version` of the table with all the node details.
- Print btn: prints the table details in `.pdf format`

gyohuangxin commented:
@theBeginner86 Good, maybe we can enable both buttons in Performance Graph page to download/print performance data in the future. This PR LGTM.

jtaleric commented:
In the Performance graph page, could you add the node-details link? That way, if there are multiple test results the user doesn't have to remember which one they just clicked to get the details.

theBeginner86 commented:
> In the Performance graph page, could you add the node-details link? That way, if there are multiple test results the user doesn't have to remember which one they just clicked to get the details.

Great UX suggestion 😃
I'll add that.

theBeginner86 commented:
https://user-images.githubusercontent.com/73700530/167829197-3286b070-5ac9-4d6a-8803-22c932307e3f.mov

Updates:
- [x] Feature to switch between views for Performance graph and node details // @jtaleric

theBeginner86 commented:
Does all changes look good now? 😅
_Ready for merge?_

theBeginner86 commented:
https://user-images.githubusercontent.com/73700530/168796831-9ea1c93c-654f-4300-ae90-5a04d9afea27.mov


updates:
- [x] Removed elevation
- [x] Added MUI tabs for switching views

theBeginner86 commented:
Now is it looking good? 😅

// @leecalcote @warunicorn19 @Abhishek-kumar09

warunicorn19 commented:
Yes looking good, now I think we can get rid of the "i" button since there are tabs we dont need another button i feel.

leecalcote commented:
You're almost there...

theBeginner86 commented:
Now removed modal title:

https://user-images.githubusercontent.com/73700530/168946092-22f90aee-1e54-40fd-981d-675cb3c979c3.mov

leecalcote commented:
@theBeginner86 good. And, title bar coloring?....

theBeginner86 commented:
ok

leecalcote commented:
@theBeginner86 yes, the charcoal green color... there should be a "standard" modal component that could be referenced for it's color.

theBeginner86 commented:
Updated color:

https://user-images.githubusercontent.com/73700530/169009184-917da714-7561-445b-a78b-6f7923b96787.mov

warunicorn19 commented:
@theBeginner86 let's merge this, i'll finish of the styling changes.
